### PR TITLE
MS-199 support different package names

### PR DESCRIPTION
**Pull request by luhmirin-s**
Notable changes:
* Adding the caller app package name to the action metadata in the orchestration flows.
* Based on the caller package name adjust the exact permission name and content provider URIs to reflect the exact caller package name.

### CODE REVIEW

File: feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/request/extractors/ActionRequestExtractor.kt [hunk 1]
```diff
@@ -24,6 +25,8 @@ internal abstract class ActionRequestExtractor(private val extras: Map<String, A
     
     open fun getBiometricDataSource(): String = extras.extractString(Constants.SIMPRINTS_BIOMETRIC_DATA_SOURCE)
 
+    open fun getCallerPackageName(): String = extras.extractString(ClientApiConstants.CALLER_PACKAGE_NAME)
+
```
Comment (BurningAXE) [inline feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/request/extractors/ActionRequestExtractor.kt:29]: Does this mean we expect the caller to send an extra "callerPackageName"?

File: feature/dashboard/src/main/java/com/simprints/feature/dashboard/debug/DebugFragment.kt [hunk 2]
```diff
@@ -47,7 +50,7 @@ internal class DebugFragment : Fragment(R.layout.fragment_debug) {
     lateinit var eventRepository: EventRepository
 
     @Inject
-    lateinit var enrolmentRecordRepository: EnrolmentRecordRepository
+    lateinit var enrolmentRecordRepository: EnrolmentRecordLocalDataSource
```
Comment (BurningAXE) [inline feature/dashboard/src/main/java/com/simprints/feature/dashboard/debug/DebugFragment.kt:53]: Not that it matters but why? :)

File: feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/request/extractors/ActionRequestExtractor.kt (hunk 1 above)
Comment (BurningAXE) [reply feature/client-api/src/main/java/com/simprints/feature/clientapi/mappers/request/extractors/ActionRequestExtractor.kt:29]: Oh, I saw the OrchestratorActivity - interesting hack :)

File: feature/dashboard/src/main/java/com/simprints/feature/dashboard/debug/DebugFragment.kt (hunk 2 above)
Comment (luhmirin-s) [reply feature/dashboard/src/main/java/com/simprints/feature/dashboard/debug/DebugFragment.kt:53]: The local data source does not need the "BiometricDataSource" arg :D It is resolved to the local source within the repo anyway.

**Pull request merged**